ext/standard/image.c

```diff
@@ -1510,27 +1528,44 @@ static void php_getimagesize_from_stream(php_stream *stream, char *input, zval *
 		case IMAGE_FILETYPE_AVIF:
 			result = php_handle_avif(stream);
 			break;
-		default:
+		default: {
+			struct php_image_handler* handler = zend_hash_index_find_ptr(&php_image_handlers, (zend_ulong) itype);
+			if (handler) {
+				result = handler->get_info(stream);
+				mime_type = handler->mime_type;
+				break;
+			}
+			ZEND_FALLTHROUGH;
+		}
 		case IMAGE_FILETYPE_UNKNOWN:
 			break;
 	}
 
 	if (result) {
-		char temp[MAX_LENGTH_OF_LONG * 2 + sizeof("width=\"\" height=\"\"")];
 		array_init(return_value);
-		add_index_long(return_value, 0, result->width);
-		add_index_long(return_value, 1, result->height);
+		if (result->width_str) {
+			add_index_str(return_value, 0, result->width_str);
+			add_index_str(return_value, 1, result->height_str);
+		} else {
+			add_index_long(return_value, 0, result->width);
+			add_index_long(return_value, 1, result->height);
+		}
```

Comment:
Is this the only case where long width and height in the result is not long? If so, I guess this could be potentially problematic for some implementation relaying on the result being int type.

Reply:
This does not affect existing uses of getimagesize(), this is unique to SVG. The reason being that in SVG you can define the width and height with a unit like "px" or "cm" etc. So in SVG I just always return it as a string.

Reply:
Yeah that's what I thought and that's actually my concern. I think it should work the same for all images and not have a different type just for SVG as it might lead to bugs if the function is used in a generic way. But loosing units is also not good. I guess it might be best to add another indexes for units.

And also how is this going to work for index 3 that should be `height="yyy" width="xxx"` that is save to add to IMG tag. I guess no units should be there either, right?

Reply:
Good points. I think adding a separate index makes sense.

> And also how is this going to work for index 3 that should be height="yyy" width="xxx" that is save to add to IMG tag. I guess no units should be there either, right?

Right... Width and height attributes are always specified in pixels, so indeed it shouldn't actually output it if we have units; it should be fine if we have the "px" unit or no units (no units defaults to px).

Reply:
Yeah agreed